### ISSUE TITLE
[Bug] Mission item progress for loiter missing

### ISSUE THREAD

**Issue by rcleaver** [PX4/PX4-Autopilot#22723]
### Describe the bug

When using a “Loiter” command (such as MAV_CMD_NAV_LOITER_TIME) in a mission, mission item progress is not streamed through telemetry.  

The old command was MAV_CMD_NAV_DELAY , MAV_CMD_NAV_LOITER_TIME  was tested but shown to be unsupported in Mission Mode. We settled on using MAV_CMD_NAV_WAYPOINT  with set at the last mission item with a loiter time. However we are still seeing the issue with each of these

### To Reproduce

1. Created mission plan as follows: 
```
1 Takeoff
2 Waypoint [some distance away]
3 Waypoint [close to last]
4 Loiter (waypoint) [at last waypoint position for 5s] - **this is what is failing to give update**
5 Waypoint [close to last]
6 Waypoint [some distance away]
7 Land
```
2. Execute mission in mission mode 
3. Telemetry streams updates for mission items correctly as expected for We get updates for mission items: 0, 1, 2, 3, 
4. No mission item updates given for mission item 4, which is loiter 
5. Telemetry resumes streaming mission item updates for 5, 6, 7. 
6. The mission reaches mission complete, but the mission progress for item 4 is wholly absent from flight logs, which is captured from telemetry streamed during execution. 

### Expected behavior

- Mission Item progress for Loiter command streams in the same manner as all other mission progress items

### Flight Log
{seq: 0, frame: 6, command: 22, current: 1, autocontinue: 1, param1: 0, param2:  0, param3:  0, param4:   0, x: 469788795, y: 319784404, z: 20, mission_type: 0}
{seq: 1, frame: 6, command: 16, current: 0, autocontinue: 1, param1: 0, param2:  1, param3:  0, param4: nan, x: 469793795, y: 319789403, z: 20, mission_type: 0}
{seq: 2, frame: 6, command: 16, current: 0, autocontinue: 1, param1: 0, param2:  1, param3:  0, param4: nan, x: 469793805, y: 319789413, z: 20, mission_type: 0}
{seq: 3, frame: 2, command: 93, current: 0, autocontinue: 1, param1: 5, param2: -1, param3: -1, param4:  -1, x:         0, y:         0, z:  0, mission_type: 0}
{seq: 4, frame: 6, command: 16, current: 0, autocontinue: 1, param1: 0, param2:  1, param3:  0, param4: nan, x: 469793814, y: 319789423, z: 20, mission_type: 0}
{seq: 5, frame: 6, command: 16, current: 0, autocontinue: 1, param1: 0, param2:  1, param3:  0, param4: nan, x: 469798794, y: 319794403, z: 20, mission_type: 0}
{seq: 6, frame: 6, command: 21, current: 0, autocontinue: 1, param1: 0, param2:  0, param3:  0, param4: nan, x: 469798794, y: 319794403, z:  0, mission_type: 0}
![image](https://github.com/PX4/PX4-Autopilot/assets/100224570/c52b9387-fb57-4858-9693-30d072ed42db)

### Software Version

PX4 v 1.13.3 
gazebo_typhoon_h480

### Flight controller

skynode

### Vehicle type

Multicopter

### Additional context

The old command was MAV_CMD_NAV_DELAY , MAV_CMD_NAV_LOITER_TIME  was tested but shown to be unsupported in Mission Mode. We settled on using MAV_CMD_NAV_WAYPOINT  with set at the last mission item with a loiter time.

However we are still seeing the issue with each of these

**Comment by julianoes** [comment]
Have you tried with v1.14.1?

**Comment by rcleaver** [comment]
> Have you tried with v1.14.1?

No, unfortunately we are locked into the other version at the moment, due to safety critical design assurance config lock for our platform

**Comment by julianoes** [comment]
I would try it with v1.14.1. If the problem is fixed, then you need to find which PR or commit fixed it and backport that for your build. If it's not fixed, we will have to look into it and fix it in main, so it will be fixed in the next release, e.g. v1.15.
